src/Tmds.Ssh/ChaCha20Decrypter.cs

```diff
@@ -0,0 +1,24 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Tmds.Ssh;
+
+static class ChaCha20Decrypter
+{
+    public static byte[] Decrypt(ReadOnlySpan<byte> key, ReadOnlySpan<byte> data)
+    {
+        // https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.chacha20poly1305
+        Span<byte> nonce = stackalloc byte[12];
+        Span<byte> tag = stackalloc byte[16];
+        tag[0] = 1;  // K_2 sets the counter to 1
+        byte[] decData = new byte[data.Length];
+
+        using var chacha = new ChaCha20Poly1305(key);
+        chacha.Encrypt(nonce, data, decData, tag);
+
+        return decData;
```

Comment:
This works in the tests but I don't have a great understanding of this protocol so I'm unsure if this is ok. I mostly just saw how the Python library asyncssh did the key decryption and this was it. They did have a verify_tag call which derived the Poly1305 key but considering we are verifying the checkint on the decrypted data I'm unsure if it is worth doing the same.

Reply:
See https://github.com/dotnet/runtime/issues/69130#issuecomment-2133268304 for some specifics about OpenSSL chacha20poly1305 vs .NET ChaCha20Poly1305.

If we're unsure, I'm fine to leave out chacha20poly1305 for now. I assume it is not used much for encrypting private keys.

Reply:
Makes sense, I think I'll just open an issue with the details from this PR so it can be revisited later on.